src/ucp/rndv/rndv.c

```diff
@@ -1752,6 +1749,8 @@ UCS_PROFILE_FUNC(ucs_status_t, ucp_rndv_rtr_handler,
          * need it registered (if am and tag is the same lane). */
         ucp_tag_offload_cancel_rndv(sreq);
         ucs_assert(!ucp_ep_use_indirect_id(ep));
+    } else if (rndv_rtr_hdr->offset == 0) {
+        ucp_request_send_untrack(sreq);
```

Comment:
how is it possible? we have nothing to send but receiver is ready to receive?

Reply:
sorry, mixed up `offset` and `remaining`
anyway, as discussed today this condition seems not enough, right?

Reply:
> anyway, as discussed today this condition seems not enough, right?

yes, I'll add TODO and comment that we should handle this for fragmented PUT, i.e. when multiple RTRs are received for a single send operation